neo/rawio/phyrawio.py

```diff
@@ -0,0 +1,259 @@
+
+  3. Step 3 : Create the neo.io class with the wrapper
+    * Create a file in neo/io/ that ends with "io.py"
+    * Create a class that inherits both your RawIO class and BaseFromRaw class
+    * copy/paste from neo/io/exampleio.py
+
+  4.Step 4 : IO test
+    * create a file in neo/test/iotest with the same previous name with "test_" prefix
+    * copy/paste from neo/test/iotest/test_exampleio.py
+
+
+
+"""
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _unit_channel_dtype,
+                        _event_channel_dtype)
+
+import numpy as np
+from pathlib import Path
+import re
+import csv
+import ast
+
+
+class PhyRawIO(BaseRawIO):
+    """
+    Class for reading Phy data.
+
+    Usage:
+        >>> import neo.rawio
+        >>> r = neo.rawio.PhyRawIO(dirname='/dir/to/phy/folder')
+        >>> r.parse_header()
+        >>> print(r)
+        >>> spike_timestamp = r.get_spike_timestamps(block_index=0,
+        ... seg_index=0, unit_index=0, t_start=None, t_stop=None)
+        >>> spike_times = r.rescale_spike_timestamp(spike_timestamp, 'float64')
+
+    """
+    extensions = []
+    rawmode = 'one-dir'
+
+    def __init__(self, dirname=''):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+        phy_folder = Path(self.dirname)
+
+        self._spike_times = np.load(phy_folder / 'spike_times.npy')
+        self._spike_templates = np.load(phy_folder / 'spike_templates.npy')
+
+        if (phy_folder / 'spike_clusters.npy').is_file():
+            self._spike_clusters = np.load(phy_folder / 'spike_clusters.npy')
+        else:
+            self._spike_clusters = self._spike_templates
+
+        # TODO: Add this when array_annotations are ready
+        # if (phy_folder / 'amplitudes.npy').is_file():
+        #     amplitudes = np.squeeze(np.load(phy_folder / 'amplitudes.npy'))
+        # else:
+        #     amplitudes = np.ones(len(spike_times))
+        #
+        # if (phy_folder / 'pc_features.npy').is_file():
+        #     pc_features = np.squeeze(np.load(phy_folder / 'pc_features.npy'))
+        # else:
+        #     pc_features = None
+
+        # SEE: https://stackoverflow.com/questions/4388626/
+        #  python-safe-eval-string-to-bool-int-float-none-string
+        if (phy_folder / 'params.py').is_file():
+            with (phy_folder / 'params.py').open('r') as f:
+                contents = f.read()
+            metadata = dict()
+            contents = contents.replace('\n', ' ')
+            pattern = re.compile(r'(\S*)[\s]?=[\s]?(\S*)')
+            elements = pattern.findall(contents)
+            for key, value in elements:
+                metadata[key.lower()] = ast.literal_eval(value)
+
+        self._sampling_frequency = metadata['sample_rate']
+
+        clust_ids = np.unique(self._spike_clusters)
+        self.unit_labels = list(clust_ids)
```

Comment:
I would use `self.unit_ids` everywhere instead of `self.unit_labels`
It will be more consistent across methods.
"label" was more an example for tridesclous than a recommandation.

Reply:
`clust_ids` are not necessarily sequential. In larger data sets some clusters have no spikes and their ids would not appear in `clust_ids` or `self.unit_labels` (e.g. 0, 1, 3, 4, 5, ...). I thought Tridesclous had `self.unit_labels` for this reason, unless I am wrong?